demisto_sdk/commands/common/tools.py

```diff
@@ -1981,3 +1981,38 @@ def suppress_stdout():
             yield
         finally:
             sys.stdout = old_stdout
+
+
+def get_definition_name(path: str, pack_path: str) -> str:
+    """
+        param:
+            path (str): path to the file which needs a definition name (generic field\generic type file)
+            pack_name (str): relevant pack name
+
+        :rtype: ``str``
+        :return:
+            for generic type and generic field return associated generic definition name folder
+
+    """
+    exception = "Generic Definition name was not found - make sure the generic field/generic type files are valid and " \
+                "connected to a generic definition file "
+    try:
+        file_dictionary = get_json(path)
+        definition_id = file_dictionary['definitionId']
+        generic_def_path = os.path.join(pack_path, 'GenericDefinitions')
+        file_names_lst = os.listdir(generic_def_path)
+        for file in file_names_lst:
+            if str.find(file, definition_id):
+                def_file_path = os.path.join(generic_def_path, file)
+                def_file_dictionary = get_json(def_file_path)
+                cur_id = def_file_dictionary["id"]
+                if cur_id == definition_id:
+                    return def_file_dictionary["name"]
+
+    except FileNotFoundError or AttributeError:
+        raise Exception(exception)
+
+    # if we couldn't find the definition name throw exception
+    raise Exception(exception)
```

Comment:
handle exceptions quietly. 
If the file was not found print something to the user like `Was unable to find the file for definitionId XXX` and return `None` or an empty string